### PR TITLE
Add health check for controller manager

### LINKED ISSUE #3288: Add health check for kserve controller manager
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Sometimes the kserve installation fails when creating cluster serving runtimes due to webhook server is not ready even though we are [waiting for the manager to be ready before applying the cluster resources](https://github.com/kserve/kserve/blob/cac5fb1407391eedee284009d31c5a45ef127b23/hack/quick_install.sh#L126). But it returns a false positive even though the webhook server is not ready. This is due to the fact that the controller manager does not implement health check endpoints.
Screenshot from 2023-11-29 16-26-33

**What did you expect to happen:**
controller manager should be in ready state after the webhook server is started.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3288

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Add health check for controller manager
```

### CODE REVIEW

**Comment by sivanantha321** [comment]
FYI:
  - kubernetes-sigs/controller-runtime#723
  - kubernetes-sigs/controller-runtime#1588

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -37,6 +37,26 @@ spec:
...
+            port: 8081
+          timeoutSeconds: 5
+        readinessProbe:
+          initialDelaySeconds: 10
+          periodSeconds: 5
+          httpGet:
+            path: /readyz
+            port: 8081
+          timeoutSeconds: 5
+        startupProbe:
```
Comment (yuzisun) [inline config/manager/manager.yaml]: Do we need startupProbe here?  The webhook should start pretty quick, here failureThreshold*periodSecond=500s which is almost 8min.

Comment (sivanantha321) [reply config/manager/manager.yaml]: Removed the startup probe. Increased readiness failureThreshold to 10. (failureThreshold*periodSecond)+initialdelay=60s. Sounds good ?

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -37,6 +37,21 @@ spec:
                 fieldPath: metadata.namespace
           - name: SECRET_NAME
             value: kserve-webhook-server-cert
+        livenessProbe:
```
Comment (yuzisun) [inline config/manager/manager.yaml:40]: Also add to the helm chart

**Pull request merged**